### PR TITLE
Add canary release action job

### PR DESCRIPTION
**Pull request by dbast**
As this package is installed into every conda installation, changes have a great impact. A canary release allows to easy test the package by a wider audience before a release is out.

This also converts the recipe:
* to be noarch: the code is noarch and it is noarch on conda-forge, noarch also allows testinng the canary build on exotic platforms like s390x where we annot easily build for via the canary action.
* to derive its package version from git history (like git describe --tags) to generate a unique increasing version per each merged PR.

The code is 1:1 from the conda repo.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-{% set version_match = load_file_regex(
-  load_file="src/conda_package_handling/__init__.py",
-  regex_pattern='^__version__ = "(.+)"') %}
-{% set version = version_match[1] %}
-
 package:
-  name: {{ name }}
-  version: {{ version }}
+  name: conda-package-handling
+  version: version: {{ GIT_DESCRIBE_TAG }}+{{ GIT_BUILD_STR }}
```
Comment (dholth) [inline conda.recipe/meta.yaml:3]: Version version?

Comment (dbast) [reply conda.recipe/meta.yaml:3]: Oh. Will fix

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
         python-version: [3.7, 3.8, 3.9]
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
```
Comment (dholth) [inline .github/workflows/tests.yml:21]: thanks

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ jobs:
         python-version: [3.7, 3.8, 3.9]
```
Comment (dholth) [inline .github/workflows/tests.yml:18]: time to add 3.10?

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (dholth) [reply conda.recipe/meta.yaml:3]: It looks like this expands to `+`only on the py37 test.

**Comment by dholth** [comment]
For some reason the build fails with `GIT_DESCRIBE_TAG is undefined`

**Pull request closed without merging**